controllers/profiles/common/mutate_visitors.go

```diff
@@ -190,8 +216,34 @@ func RolloutDeploymentIfCMChangedMutateVisitor(workflow *operatorapi.SonataFlow,
 	return func(object client.Object) controllerutil.MutateFn {
 		return func() error {
 			deployment := object.(*appsv1.Deployment)
+			restoreKnativeVolumeAndVolumeMount(deployment)
 			err := kubeutil.AnnotateDeploymentConfigChecksum(workflow, deployment, userPropsCM, managedPropsCM)
 			return err
 		}
 	}
 }
+
+func moveKnativeVolumeToEnd(vols []corev1.Volume) {
+	for i := 0; i < len(vols)-1; i++ {
+		if vols[i].Name == KnativeBundleVolume {
+			vols[i], vols[i+1] = vols[i+1], vols[i]
+		}
+	}
+}
+
+func moveKnativeVolumeMountToEnd(mounts []corev1.VolumeMount) {
+	for i := 0; i < len(mounts)-1; i++ {
+		if mounts[i].Name == KnativeBundleVolume {
+			mounts[i], mounts[i+1] = mounts[i+1], mounts[i]
+		}
+	}
+}
+
+// Knative Sinkbinding injects K_SINK env, a volume and volumn mount. The volume and volume mount
+// must be in the end of the array to avoid repeadly restarting of the workflow pod
+func restoreKnativeVolumeAndVolumeMount(deployment *appsv1.Deployment) {
+	moveKnativeVolumeToEnd(deployment.Spec.Template.Spec.Volumes)
+	for i := 0; i < len(deployment.Spec.Template.Spec.Containers); i++ {
+		moveKnativeVolumeMountToEnd(deployment.Spec.Template.Spec.Containers[i].VolumeMounts)
+	}
+}
```

Comment:
Please move these knative specifics to a `knative.go` file in this module.

Reply:
Done